components/FormikFields/PaperInputPicker/AutoFill/index.js

```diff
@@ -22,9 +23,11 @@ export default class AutoFill extends Component {
 
   componentDidMount() {
     const { parameter } = this.props;
-    retrievePuenteAutofillData(parameter)
-      .then((data) => {
-        this.state.fields = data;
+    cacheAutofillData(parameter)
+      .then(async () => {
+        const data = await getData('autofill_information');
+        const result = data[parameter];
+        this.state.fields = result;
```

Comment:
hmm this is technically a react anti-pattern

[Check this out](https://stackoverflow.com/questions/35867038/what-the-difference-of-this-state-and-this-setstate-in-reactjs)

when mutating the state in a class-based component, you usually use `setState`